keps/sig-scheduling/3902-decoupled-taint-manager/README.md

```diff
@@ -0,0 +1,292 @@
+            AddFunc: controllerutil.CreateAddNodeHandler(func(node *v1.Node) error {
+                nc.taintManager.NodeUpdated(nil, node)
+                return nil
+            }),
+            UpdateFunc: controllerutil.CreateUpdateNodeHandler(func(oldNode, newNode *v1.Node) error {
+                nc.taintManager.NodeUpdated(oldNode, newNode)
+                return nil
+            }),
+            DeleteFunc: controllerutil.CreateDeleteNodeHandler(func(node *v1.Node) error {
+                nc.taintManager.NodeUpdated(node, nil)
+                return nil
+            }),
+        })
+    }
+        if err != nil {
+        return nil, true, fmt.Errorf("failed to start the taint manager: %v", err)
+    }
+  }  
+```
+
+A feature gate `SeparateTaintManager` controls whether to use the split `TaintManager` or  roll back to the old `NodeLifecycleController`.
+
+### Test Plan
+
+[x] I/we understand the owners of the involved components may require updates to existing tests to make this code solid enough prior to committing the changes necessary to implement this enhancement.
+
+##### Prerequisite testing updates
+
+Kubernetes already has a good coverage of node `No-Execute` eviction. We will add tests only for the changed code.
+
+#####  Unit tests
+- `pkg/controller/taintmanager`: 2023-05-23 - 0%
+- `pkg/controller/apis/config/v1alpha1`: 2023-05-23 - 0%
+
+#####  Integration tests
+ No integration tests are required. This feature is better tested with e2e tests.  
+
+##### E2E tests
+
+- Verify passing the existing E2E and conformance tests using the split `TaintManager`.
+- Verify the ability to disable the default `TaintManager`.
+
+### Graduation Criteria
+
+#### Alpha
+
+* Support `kube-controller-manager` flag `--controllers=-taint-manager` to disable the default taint-manager.
+* Unit and integration tests completed and passed.
+
+#### Beta
+
+* Performance and scalability tests to verify there is non-negligible performance degradation in node taint-based eviction.
+* Update documents to reflect the changes.
+
+#### GA
+
+* Fix all reported bugs if any.
+
+### Upgrade / Downgrade Strategy
+
+We’ll need to upgrade `kube-controller-manager` and` node-lifecycle-controller` and add a new controller `taint-manager`. Accordingly, a downgrade will revert to the previous version of `kube-controller-manager` and `node-lifecycle-controller` and remove the new added `taint-manager` . There should be no impact on the other components during upgrades and downgrades since user-exposed behavior should stay the same except a new flag to enable/disable the default `taint-manager`. This change should be clearly communicated to users.
+
+### Version Skew Strategy
+
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+*This section must be completed when targeting alpha to a release.*
+
+* **How can this feature be enabled / disabled in a live cluster?**
```

Comment:
Mention the feature gate

Reply:
Please update to the new KEP template (as mentioned above) - the feature-gate is then required explicitly :)